cli/internal/run/run.go

```diff
@@ -497,7 +497,11 @@ func (c *RunCommand) Run(args []string) int {
 
 				// Setup command execution
 				argsactual := append([]string{"run"}, task)
-				argsactual = append(argsactual, runOptions.passThroughArgs...)
+				for _, target := range ctx.Targets {
+					if target == task {
+						argsactual = append(argsactual, runOptions.passThroughArgs...)
```

Comment:
I think we may want to do this check earlier in the process. I suspect we'll get better cache utilization if we aren't hashing args that we aren't using for dependent tasks.

What about calculating this closer to [here](https://github.com/vercel/turborepo/blob/main/cli/internal/run/run.go#L438)